software/acpid.txt

```diff
@@ -0,0 +1,169 @@
+    - Handling Events                                                      [3.2]
+
+
+[0.0] Usage
+________________________________________________________________________________
+
+KISS Linux offers two options for acpid management: busybox acpid, which is
+installed by default, and acpid2, which can be installed from the acpid
+package. For both acpid implimentations, you will need to enable a few
+ACPI-related kernel options and optionally the runit service to manage it.
+
+To enable the acpid runit service run:
+
++------------------------------------------------------------------------------+
+|                                                                              |
+|   $ ln -s /etc/sv/acpid/ /var/service/                                       |
+|   $ sv up acpid                                                              |
+|                                                                              |
++------------------------------------------------------------------------------+
+
+Whenever changes are made, restart acpid for them to take effect.
+
++------------------------------------------------------------------------------+
+|                                                                              |
+|   $ sv restart acpid                                                         |
+|                                                                              |
++------------------------------------------------------------------------------+
+
+
+[1.0] Kernel Setup
+________________________________________________________________________________
+
+ACPI-related kernel drivers must be enabled for acpid to function properly. In
+menuconfig, these options are found under Power management and ACPI options
+> Power Management support > ACPI support. Most of the drivers are
+self-explanatory, but the following notable options can be disabled:
+
++------------------------------------------------------------------------------+
+|                                                                              |
+|   CONFIG_ACPI_PROCFS_POWER    This option is deprecated                      |
+|   CONFIG_ACPI_EC_DEBUGFS      Potentially interferes with reboot             |
+|   CONFIG_ACPI_TABLE_UPGRADE   KISS does not use an initrd by default         |
+|   CONFIG_ACPI_DEBUG           Adds 50k to kernel size                        |
+|   CONFIG_ACPI_PCI_SLOT        Usually unnecessary                            |
+|   CONFIG_ACPI_CUSTOM_METHOD   Potential security flaw                        |
+|                                                                              |
++------------------------------------------------------------------------------+
+
+
+[2.0] Busybox acpid
+________________________________________________________________________________
+
+Busybox provides a very limited acpid program, that only works well for laptop
+lid events. Every time the lid is closed, acpid will execute
```

Comment:
Is this claim correct? The source code seems to include more events: https://git.busybox.net/busybox/tree/util-linux/acpid.c#n61

Reply:
The others don't work. Hence, "only works well for laptop lid."  The other macros that are given aren't actually used anywhere else in the program. Its possible to define new keys in /etc/acpi.map and /etc/acpid.conf with raw keycodes, but the last time I looked into it there was very little documentation and I couldn't get it to work. I can look into it again and add something later if I figure it out.